integration-test/src/main/java/com/sequenceiq/it/cloudbreak/NetworksTest.java

```diff
@@ -0,0 +1,204 @@
+package com.sequenceiq.it.cloudbreak;
+
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.ForbiddenException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Optional;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import com.sequenceiq.cloudbreak.api.model.CredentialResponse;
+import com.sequenceiq.cloudbreak.api.model.PlatformResourceRequestJson;
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakClient;
+import com.sequenceiq.it.cloudbreak.newway.CloudbreakTest;
+import com.sequenceiq.it.cloudbreak.newway.Credential;
+import com.sequenceiq.it.cloudbreak.newway.Networks;
+import com.sequenceiq.it.cloudbreak.newway.TestParameter;
+import com.sequenceiq.it.cloudbreak.newway.cloud.CloudProvider;
+import com.sequenceiq.it.cloudbreak.newway.cloud.CloudProviderHelper;
+import com.sequenceiq.it.cloudbreak.newway.cloud.OpenstackCloudProvider;
+import com.sequenceiq.it.util.LongStringGeneratorUtil;
+
+public class NetworksTest extends CloudbreakTest {
+
+    private static final String VALID_CRED_NAME = "valid-cred-";
+
+    private static final String INVALID_CRED_NAME = "invalid-cred";
+
+    private static final String INVALID_AV_ZONE = "invalid-avzone";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NetworksTest.class);
+
+    private String credentialName = "";
+
+    private Long credentialId = 0L;
+
+    private CloudProvider cloudProvider;
+
+    @Inject
+    private LongStringGeneratorUtil longStringGeneratorUtil;
+
+    public NetworksTest() {
+    }
+
+    public NetworksTest(CloudProvider cp, TestParameter tp) {
+        cloudProvider = cp;
+        setTestParameter(tp);
+    }
+
+    @BeforeTest ()
+    @Parameters("provider")
+    public void beforeTest(@Optional(OpenstackCloudProvider.OPENSTACK) String provider) {
+        LOGGER.info("before cluster test set provider: " + provider);
+        if (cloudProvider != null) {
+            LOGGER.info("cloud provider already set - running from factory test");
+            return;
+        }
+        cloudProvider = CloudProviderHelper.providerFactory(provider, getTestParameter())[0];
+    }
+
+    @BeforeTest()
+    public void createCredentialsAllProviders() throws  Exception {
+        credentialName = VALID_CRED_NAME + cloudProvider.getPlatform().toLowerCase() + "-nets";
+
+        given(CloudbreakClient.isCreated());
+        given(cloudProvider.aValidCredential()
+                .withName(credentialName)
+                .withDescription("")
+                .withCloudPlatform(cloudProvider.getPlatform()), credentialName + " credential is created.");
+        when(Credential.get(), credentialName + " credential has been got.");
```

Comment:
We can spare the when/than steps in a BeforeTest method. I mean we have sufficient tests for credential, and the credential here should be "given"; the when then steps can focus on the network related things.